lib/ae_page_objects/element_proxy.rb

```diff
@@ -16,31 +16,21 @@ def initialize(element_class, *args)
     end
 
     def visible?(options = {})
-      wait_until_visible(options[:wait])
-      true
-    rescue ElementNotVisible
-      false
+      reload_element
```

Comment:
Will this not raise LoadElementFailed in element.rb line 129 if the element is not visible?

Similar for the other predicates. I think the test stub behaves different from the real Element.rb in element_proxy_test.rb

Reply:
There are 3 possible cases:

1. **The element exists and is visible.**
    The element will be loaded and the method will return `true`.

2. **The element exists and is not visible**:
    The element will be loaded and the method will return `false`.

3. **The element does not exist**:
    The constructor of `Element` will raise `LoadElementFailed`. This is caught in `ElementProxy#reload_element`, which will result in `@loaded_element = nil`. The `visible?` method would then return `false`.

Reply:
My bad, I missed the rescue block.